plugin/src/main/java/org/autorefactor/refactoring/ASTBuilder.java

```diff
@@ -857,4 +864,37 @@ public Expression parenthesizeIfNeeded(Expression expr) {
             return expr;
         }
     }
+
+    /**
+     * Builds a new {@link Statement} instance which is basically a newline.
+     *
+     * @return a newline statement
+     */
+    public Statement newlinePlaceholder() {
+        return (Statement) this.refactorings.getRewrite().createStringPlaceholder("\n", ASTNode.EMPTY_STATEMENT);
+    }
+
+    public Modifier modifierProtected() {
+        return getAST().newModifier(ModifierKeyword.PROTECTED_KEYWORD);
+    }
+
+    public List<IExtendedModifier> modifiers(IExtendedModifier... modifiers) {
+        return Arrays.asList(modifiers);
+    }
+
+    public Expression superInvoke(String methodName) {
+        SuperMethodInvocation superMethodInvocation = getAST().newSuperMethodInvocation();

```

Comment:
I do not think I am the only one.
I have some references to point out:
http://www.itiseezee.com/?p=83

> The length of a name should correspond to the size of its scope.
> Single-letter names should ONLY be used as local variables inside short methods.

Of course you may use other advices in this web page to counteract what I am saying :)

&nbsp;


I am always in favour of descriptive names.
However here I do not think the long name adds any value. I get bored before I can read the whole `superMethodInvocation`, and I can easily remember that `smi` is shorthand for `superMethodInvocation` in these 3 lines of code. I am even tempted to just name this variable `result` :)

&nbsp;

I can show you places where I have chosen very long names and I will not change them because this long name helps make the code more readable IMO:
https://github.com/JnRouvignac/AutoRefactor/blob/master/plugin/src/main/java/org/autorefactor/refactoring/ASTHelper.java#L1443

&nbsp;

Anyway, this is my POV, you may disagree :)